src/python/pants/backend/docker/goals/package_image_test.py

```diff
@@ -272,6 +272,11 @@ def test_build_docker_image(rule_runner: RuleRunner) -> None:
                   name="test5",
                   image_tags=["latest", "alpha-1.0", "alpha-1"],
                 )
+                docker_image(
+                  name="test6",
+                  image_tags=["1.2.3"],
+                  repository="xyz/{build_file_path}/{name}",
```

Comment:
Is there particular motivation for adding this to the interpolation context, rather than using `build_file_dir`?
```suggestion
                  repository=f"xyz/{build_file_dir()}/{{name}}",
```

The one case where the above won't work is when providing a default repository value using the `[docker].default_repository` or on the registry config section. The former can be addressed by using `__defaults__`, which leaves the option to configure this per registry.

Reply:
No motivation either way. I do not know the code base, or what was/is in the interpolation context. so was unaware. 

Are you suggesting @kaos, that the `build_file_dir` needs to be "blessed" to be available for this format string?
That is, in order for a user to do what you suggest above, is there code change still required somewhere ? 

The original motivation is this thread on slack, 
https://pantsbuild.slack.com/archives/C046T6T9U/p1707568944184539
where I needed the full path directory to the BUILD file. 
as it is, you can only get the current BUILD file `directory` or it's immediate `parent_directory` and no higher. 

so 
`dir1/dir2/BUILD` - i can obtain `dir1` & `dir2`  (`parent_directory` and `directory`)
but
`dir0/dir1/dir2/BUILD` - i cannot obtain `dir0` via a variable (for use either in `BUILD` or `pants.toml`)

If there _is_ a way already, then I do not wish for any PR :-D

Reply:
> The former can be addressed by using `__defaults__`,

Hm, my initial reaction would be that `__defaults__({docker_image: dict(repository=f"{build_file_dir()}"})`, would use the BUILD file dir that contains the `__defaults__`, not the one that contains each `docker_image` that uses that default?

I think this _can_ definitely be currently addressed by a macro wrapping `docker_image`, but that's pretty annoying for users, especially given many of the other existing interpolation values can be handled via macro in a similar way, so it feels a bit unfortunate to exclude this one.

Reply:
True, the `__defaults__` does not work. My mistake.

Apologies if I came across as overly against this change, I am not. I was just lacking the context. (I've read up on the slack thread now, so this makes much more sense to me now.)

Will do a proper review.

Reply:
Not a problem - alls good / all normal.